app/concerns/koudoku/subscription.rb

```diff
@@ -194,6 +194,10 @@ def update_default_stripe_card
     customer = Stripe::Customer.retrieve(self.stripe_id)
     source = customer.sources.create(source: credit_card_token)
     customer.default_source = source.id
+    # Undo payment_method set by the payment_methods API as this will override the default_source
+    if !customer.invoice_settings.default_payment_method.nil?
+      customer.invoice_settings.default_payment_method = nil
+    end
```

Comment:
This looks like a good fail-safe to ensure the default gets updated on the customer.